### PR TITLE
[BEAM-10422] Chicago Taxi example: set number of Dataflow workers to 5

### PR DESCRIPTION
**Pull request by kamilwu**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | --- | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Python PreCommit

**Comment by kamilwu** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
R: @tvalentyn 

This effectively fixes https://issues.apache.org/jira/browse/BEAM-10421 as well. Look for "worker logs:" lines, there are links to Dataflow job details

**Comment by kamilwu** [comment]
@tvalentyn Would you mind taking a look?

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy [hunk 1]
```diff
@@ -19,46 +19,44 @@
...
             switches('-PgcsRoot=gs://temp-storage-for-perf-tests/chicago-taxi')
+            switches("-PpipelineOptions=\"${LoadTestsBuilder.parseOptions(pipelineOptions)}\"")
         }
     }
 }
 
+PostcommitJobBuilder.postCommitJob(
+    'beam_PostCommit_Python_Chicago_Taxi_Dataflow',
+    'Run Chicago Taxi on Dataflow',
+    'Google Cloud Dataflow Runner Chicago Taxi Example',
```
Comment (tvalentyn) [inline .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:50]: It's nice to include trigger command in the message (easy to rerun if needed). How about:

Chicago Taxi Example on Dataflow (Run Chicago Taxi on Dataflow)

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy [hunk 2]
```diff
@@ -19,46 +19,44 @@
...
+        autoscaling_algorithm: 'NONE',
+    ]
 
     // Gradle goals for this job.
-    steps {
+    scope.steps {
         gradle {
             rootBuildScriptDir(commonJobProperties.checkoutDir)
+            commonJobProperties.setGradleSwitches(delegate)
             tasks(':sdks:python:test-suites:dataflow:py2:chicagoTaxiExample')
```
Comment (tvalentyn) [inline .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:40]: Can we switch this to Py3?

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy [hunk 3]
```diff
@@ -19,46 +19,44 @@
...
+PostcommitJobBuilder.postCommitJob(
+    'beam_PostCommit_Python_Chicago_Taxi_Dataflow',
+    'Run Chicago Taxi on Dataflow',
+    'Google Cloud Dataflow Runner Chicago Taxi Example',
+    this
+) {
+    chicagoTaxiJob(delegate)
+}
+
 CronJobBuilder.cronJob(
```
Comment (tvalentyn) [inline .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:56]: With CronJobBuilder and PostcommitJobBuilder, do we run these test as frequent as specified by Cron config or more frequently (as often as new commits come in) ?

Comment (kamilwu) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:56]: As frequent as specified by Cron only. 

https://github.com/apache/beam/blob/0fc74c09f1b90c1b954b78973e83483daa51fd18/.test-infra/jenkins/CronJobBuilder.groovy#L47

The fourth parameter, `triggerOnCommit`, is set to False here, which means that the job is not triggered when new commits come in.

File: .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy (hunk 2 above)
Comment (kamilwu) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:40]: There's a jira for this: https://issues.apache.org/jira/browse/BEAM-9154. As far as I remember there was something that prevented us from switching to Py3. Not sure if it's still relevant.

Comment (tvalentyn) [reply .test-infra/jenkins/job_PostCommit_Python_Chicago_Taxi_Example_Dataflow.groovy:40]: We should move this suite to Py3 as we are actively discussing dropping Py2 support, and these suites will stop working.

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Chicago Taxi on Dataflow

**Comment by kamilwu** [comment]
Thanks @tvalentyn. I agree we should move this suite to Py3, probably even as a part of  https://issues.apache.org/jira/browse/BEAM-8892 (move Chicago Taxi to new Beam orchestrator developed by the TFX team). I will merge this PR to keep the test working until we move it to Py3.

**Pull request merged**